### PR TITLE
Adds particle visuals to some of Defiler's abilities

### PR DESCRIPTION
**Pull request by lKiseki**
## About The Pull Request
Adds particles to the Defiler's Reagent Slash and Emit Gas abilities, visually enhancing them. Proof of concept below.
Also ports particle holders from TG, necessary for these to function correctly.

**REAGENT SLASH:**

https://user-images.githubusercontent.com/59634950/193898550-638aac7a-101d-4188-b476-0ef0f025e316.mp4

**EMIT NEUROGAS:**

https://user-images.githubusercontent.com/59634950/193898547-1eee7de2-e274-4adb-a2e1-1666b77543c4.mp4

Note that I used reagent slash for this just for show. In practice, it looks like so:

https://user-images.githubusercontent.com/59634950/193481315-afcab5ef-8471-4546-8f91-bdf8e9fb6713.mp4

## Why It's Good For The Game
Game look pretty. Therefore, game good.

## Changelog
:cl: Lewdcifer
imageadd: Added particle visuals to Defiler's Reagent Slash and Emit Neurogas abilities.
/:cl:

### CODE REVIEW

**Comment by lKiseki** [comment]
Someone in the Discord expressed a concern regarding whether particles were visible over smoke. This is false. Proof below:

https://user-images.githubusercontent.com/59634950/193481384-b2acfba3-27ce-4d8a-8878-d4fc75df760f.mp4

**Review by Kuro020 — APPROVED** [review]

**Comment by novaepee** [comment]
pretty

**Comment by 567Turtle** [comment]
thank you very c

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 1]
```diff
@@ -177,7 +244,27 @@
 		T.visible_message(span_danger("Noxious smoke billows from the hulking xenomorph!"))
 		count = max(0,count - 1)
 		sleep(DEFILER_GAS_DELAY)
+	toggle_particles(TRUE)
 
+// Toggles particles on or off, depending on the defined var.
+/datum/action/xeno_action/emit_neurogas/proc/toggle_particles(deactivate)
+	var/mob/living/carbon/xenomorph/X = owner
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:251]: please invert these (ie deactivate becomes activate)
Usually toggles mean TRUE = ON, FALSE = OFF and this is inverted

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 2]
```diff
@@ -378,6 +468,29 @@
 	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
 	return ..()
 
+// Toggles particles on or off, depending on the defined var.
+/datum/action/xeno_action/reagent_slash/proc/toggle_particles(deactivate)
+	var/mob/living/carbon/xenomorph/X = owner
+
+	if(deactivate)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:475]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**